File: src/test/java/com/ysmull/easeshop/TestService.java
package com.ysmull.easeshop;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Isolation;
import org.springframework.transaction.annotation.Transactional;

import static org.junit.Assert.assertEquals;

@Service
public class TestService {

    @Autowired
    TestDao testDao;

    @Transactional(isolation = Isolation.REPEATABLE_READ)
    public void sqlTest() {
        testDao.insert();
        long id = testDao.selectId();
        String actualName = testDao.selectName(id);
        assertEquals(Thread.currentThread().toString(), actualName);
    }
}

File: EaseMall-web/src/test/java/com/ysmull/easemall/TestService.kt
package com.ysmull.easemall

import org.junit.Assert.assertEquals
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Isolation
import org.springframework.transaction.annotation.Transactional

@Service
class TestService(
        internal var testDao: TestDao
) {
    @Transactional(isolation = Isolation.REPEATABLE_READ)
    fun sqlTest() {
        testDao.insert()
        val id = testDao.selectId()
        val actualName = testDao.selectName(id)
        assertEquals(Thread.currentThread().toString(), actualName)
    }
}
